### DISCUSSION TITLE
Need help for map variable from a database writer

### DISCUSSION THREAD

**Discussion by postheimer** [nextgenhealthcare/connect#5202] [Q&A] [answered]
Hello 

i Have a channel with a source is a trigger file reader in a folder wich have pdf 

in the source i have a transformer for extracting a number from the filename 

after in destinations i have 3 destinations , the first move the pdf in a smb destination 

the second is a database writer  with a select from a database wich extract several information for the last destinations

and here is the problem 

the response is database write sucess  so the sql query is OK  but i want use the SQL response to a variable for the last destination

i have test with transformer  to map a variable with exemple ( msg['pbmrn'].ToString() ) but the variable is null on the last destination

where is my error ?

**Comment by tonygermano** [comment, marked as answer]
A Database Sender executes an update, so it returns the rows affected rather than a result set. You'll probably have to do your select statement using javascript, in which case you can likely combine the 2nd and 3rd destinations.

**Reply by postheimer** [reply]
i confirm with a javascript writer that works 

thanks
